pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -486,7 +486,16 @@ public Boolean allowNamespacePolicyOperation(NamespaceName namespaceName, Policy
         }
 
         if (provider != null) {
-            return provider.allowTopicOperationAsync(topicName, originalRole, role, operation, authData);
+            switch (operation) {
+                case PRODUCE:
+                    return canProduceAsync(topicName, role, authData);
+                case CONSUME:
+                    return canConsumeAsync(topicName, role, authData, null);
+                case LOOKUP:
+                    return canLookupAsync(topicName, role, authData);
+                default:
+                    return provider.allowTopicOperationAsync(topicName, originalRole, role, operation, authData);
```

Comment:
I think it is better to change the default implementation in the AuthorizationProvider interface instead of here. 

```
    /**
     * Grant authorization-action permission on a topic to the given client
     * @param topic
     * @param originalRole role not overriden by proxy role if request do pass through proxy
     * @param role originalRole | proxyRole if the request didn't pass through proxy
     * @param operation
     * @param authData
     * @return CompletableFuture<Boolean>
     */
    default CompletableFuture<Boolean> allowTopicOperationAsync(TopicName topic, String originalRole, String role,
                                                             TopicOperation operation,
                                                             AuthenticationDataSource authData) {
switch (operation) {
                case PRODUCE:
                    return canProduceAsync(topicName, role, authData);
                case CONSUME:
                    return canConsumeAsync(topicName, role, authData, null);
                case LOOKUP:
                    return canLookupAsync(topicName, role, authData);
                default:
                     return FutureUtil.failedFuture(new IllegalStateException("TopicOperation is not supported by the Authorization provider you are using."));
    }
```

Reply:
@sijie Done. PTAL